src/containers/apps/appDetails/deploy/Deployment.tsx

```diff
@@ -25,14 +25,27 @@ export default class Deployment extends ApiComponent<
       | undefined;
   }
 > {
-  constructor(props: any) {
+  initRepoInfo: string;
+
+  constructor(props: AppDetailsTabProps) {
     super(props);
     this.state = {
       dummyVar: undefined,
       forceEditableCaptainDefinitionPath: false,
       updatedVersions: undefined,
       buildLogRecreationId: ""
     };
+
+    const { appPushWebhook } = props.apiData.appDefinition;
```

Comment:
extracting props in the render method don't work for this. 
Then there is nothing to compare (always equal). I think alternative to do this in constructor I can put it in componentDidMount.

this could be an alternative solution to avoid changes.
```ts
    this.initRepoInfo = JSON.stringify(appPushWebhook 
      ? {...appPushWebhook.repoInfo} 
      : {
          user: "",
          password: "",
          branch: "",
          sshKey: "",
          repo: ""
    });
```